samples/cellular/nrf_cloud_multi_service/prj.conf

```diff
@@ -94,6 +94,8 @@ CONFIG_MODEM_INFO_ADD_NETWORK=y
 CONFIG_NRF_CLOUD_PGPS=y
 CONFIG_NRF_CLOUD_PGPS_REPLACEMENT_THRESHOLD=4
 CONFIG_NRF_CLOUD_PGPS_REQUEST_UPON_INIT=y
+CONFIG_FOTA_DOWNLOAD_ADDR_FAM_IPV4=y
```

Comment:
This will make the default build of this sample incompatible with IPv6, right?

If we have specific test hardware that needs IPV4, I think we should set the builds for that hardware to use IPV4 rather than apply this as the sample default

But I have another concern: The `_UNSPEC` option should, at least in theory, try both IPv6 and IPv4, right?

So I do not see how this will solve the issue we've been seeing with IPv6 connection failures -- In theory those failed IPv6 downloads should have fallen back to IPv4 on their own.

Reply:
I was unware that we are now seeing carriers with IPv6-only networks. I think in the past (an NCS rev or two in the past) our samples were set to IPv4 only.

If IPv6-only networks are now common, I wonder why we have not been getting complaints about not being able to use nrfcloud libraries in older NCS versions?

Reply:
> I was unware that we are now seeing carriers with IPv6-only networks.

It was not my intent to imply this, just to be clear. I don't know how common IPv6 is as a requirement.

But it is true that networks can, at any time, impose IPv6 on the PDN. So even if they are not doing it today, it seems within the realm of possibility to me that networks might soon start doing so.

That, coupled with the fact that we already have an IPv4/IPv6 fallback mechanism that seems to work 99% of the time, makes me hesitate to approve changing the defaults for our samples.

Have we reproduced the issue with IPv6 at any point? Last I checked in on it, we had attempted to reproduce it, but couldn't, because the cell networks we kept trying were enforcing IPv4. So until we can reproduce on an IPv6-capable network, I am furthermore hesitant to disable IPv6

Reply:
Maybe this logic from Stig:

> The changes in commit https://github.com/nrfconnect/sdk-nrf/commit/57c5706f8f2ac13d5c1e0d064fdad27f9f9e2e04 fixed two issues where using AF_UNSPEC also would fail. One when the network does not give the device an IPv6 address (happens in many networks and when using MVNO SIMs), and another when the network does not correctly route IPv6 traffic (this happens for one of the big US carriers).

needs to be repeated everywhere we try to connect to a socket, not just in the download client.

Reply:
But I also don't want to steamroll -- I will retract my request for changes if you are still not convinced by the above

Reply:
> Maybe this logic from Stig needs to be repeated everywhere we try to connect to a socket, not just in the download client.

Yes, or perhaps integrated directly into the nrf91 offloaded socket interface (I ***think*** that would be possible)